.github/workflows/pre_commit.yml

```diff
@@ -0,0 +1,22 @@
+name: Pre-Merge Checks
```

Comment:
Should this file be named as `pre_merge.yaml`? I guess we will add pre-merge checks here as well?

Reply:
No my bad. I think we can have it in the same workflow file